_studio/mfx_lib/scheduler/include/mfx_scheduler_core.h

```diff
@@ -395,12 +394,12 @@ class mfxSchedulerCore : public MFXIScheduler2
     // Dependency table.
     // There are an index: it holds the number of values valid in the array.
     // The array may have larger size.
-    UMC::Array<MFX_DEPENDENCY_ITEM> m_pDependencyTable;
+    std::vector<MFX_DEPENDENCY_ITEM> m_pDependencyTable;
     volatile
     mfxU32 m_numDependencies;
```

Comment:
Why not to use m_pDependencyTable.size() instead of  "volatile mfxU32 m_numDependencies"

Reply:
work with m_pDependencyTable designed as for const size array.
I guess we shouldn't rework current behavior in this PR

Reply:
Why volatile, if it is size of const size array? I'd rather see
constexpr size_t m_numDependencies = MFX_MAX_NUMBER_TASK * 2;

Reply:
m_numDependencies - is actual number of dependency (<= MFX_MAX_NUMBER_TASK * 2)

Reply:
Ok, you are right, it is possible to leave it for future.
Anyway, while array size is m_pDependencyTable.capacity() equal to MFX_MAX_NUMBER_TASK * 2, using m_pDependencyTable.size() instead of m_numDependencies would be more clear.
And it is related to the change. Better do it now, I think.